### DISCUSSION TITLE
Associated work items not being pulled from pull request? Am I doing something wrong?

### DISCUSSION THREAD

**Discussion by cmw9706** [rfennell/AzurePipelines#1878] [Q&A]
I am trying to create release notes from work items associated to a pull request in ADO.

I'm just testing right now but I have a pipeline to trigger when I make a PR against my test branch. I have the dump to console option set to true and when I make a PR with an associated work item, the `workitems` returns as
```
 "workItems": [],
```
I also see

```
"associatedWorkitems": [],
```

I checked the api and it looks like it should return the work item

GET:

https://thycotic.visualstudio.com/8118cbb3-5d8b-40bc-83d9-4481c26cbb0e/_apis/git/repositories/51f9f61f-4c0c-43eb-bef3-c2a366131d84/pullRequests/114601/workitems

RESULT:

```json 
{"count":1,"value":[{"id":"564717","url":"https://thycotic.visualstudio.com/_apis/wit/workItems/564717"}]}
```
Am I doing something wrong? I would expect these to be populated with my associated work items in ado.

**Comment by rfennell** [comment]
The task is using the Azure DevOps API calls

- For a build that has never succeeded for a pipeline/banch/stage
   - https://learn.microsoft.com/en-us/rest/api/azure/devops/build/builds/get-build-changes?view=azure-devops-rest-7.2
   - https://learn.microsoft.com/en-us/rest/api/azure/devops/build/builds/get-build-work-items-refs?view=azure-devops-rest-7.2
- Subsequent builds
   - https://learn.microsoft.com/en-us/rest/api/azure/devops/build/builds/get-changes-between-builds?view=azure-devops-rest-7.2
   - https://learn.microsoft.com/en-us/rest/api/azure/devops/build/builds/get-work-items-between-builds?view=azure-devops-rest-7.2

So it is these API calls you need to test with.

I would suggest testing using the [local tester](https://github.com/rfennell/AzurePipelines/tree/main/Extensions/XplatGenerateReleaseNotes/XplatGenerateReleaseNotesTask/testconsole) as this allow repeated running, and default to giving a detailed log (which can also be show with the variable `system.debug=true`)

Upload a copy of  detailed log if it is not obvious why it is not giving the expected results
